File: src/main/kotlin/br/com/caethas/atto/atto/modules/membro/controller/MembroController.kt
package br.com.caethas.atto.atto.modules.membro.controller

import br.com.caethas.atto.atto.modules.membro.dto.MembroDto
import br.com.caethas.atto.atto.modules.membro.entity.MembroEntity
import br.com.caethas.atto.atto.modules.membro.repository.MembroRepository
import br.com.caethas.atto.atto.modules.membro.service.MembroService
import br.com.caethas.atto.atto.shared.base.BaseController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/membros")
class MembroController(service: MembroService) : BaseController<MembroEntity, MembroRepository, MembroDto>(service) {
}

File: lib/src/modules/membro/membro_controller.dart
import 'package:atto_api/src/modules/membro/dto/membro_dto.dart';
import 'package:atto_api/src/modules/membro/entity/membro_entity.dart';
import 'package:atto_api/src/modules/membro/service/membro_service_interface.dart';
import 'package:atto_api/src/shared/controller/base_controller.dart';
import 'package:vaden/vaden.dart';

@Api(tag: 'Membro', description: 'Gerenciamento de membros e requisições relacionadas a membros.')
@Controller('/membros')
class MembroController extends BaseController<MembroDto, MembroEntity, IMembroService> {

  MembroController(IMembroService service) : super(service);

  @Get('/')
  Future<Response> getAll() => super.getAll();

  @Get('/<id>')
  Future<Response> getById(@Param() String id) => super.getById(id);

  @Post('/')
  Future<Response> create(@Body() MembroDto membro) async {
    final entity = entityFromDto(membro);
    return await super.save(entity);
  }
}